### PR TITLE
Docs: fix broken links

### PR DESCRIPTION
**Pull request by clementduveau**
**What is this feature?**

Not a feature, this bring fix to different links missing/broken in the docs

**Why do we need this feature?**

Easier onboarding, better self-serve experience of course !

**Who is this feature for?**

Any customer or OSS admin

**Which issue(s) does this PR fix?**:

I did not find these from issues

Fixes:

- Difference between legacy / unified alerting was linked several time but page was in draft. As it's now default, references are fading away with v9 and v8 but still relevant for those stuck or trying to upgrade. The page will be available but not listed. Referenced is pinned to v10 of the docs to ensure that it stays functional in the future.
- typo in a link to correlations API
- troubleshooting links to /docs/grafana/latest/age-encryption.org instead of https://age-encryption.org, link is fixed
- "What's-new v8.2" links to "latest unified-alerting" (that became "alerting"). Fixed by pinning to the version 8.2
- "What's-new v8.5" links to a page on how to enable service accounts in latest version, that doesn't exist. Pinned to v8.5 instead (the intended page).

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by imatwawana — COMMENTED** [review]
@brendamuir - I think you'll want to review the changes to the `docs/sources/alerting/difference-old-new.md file.`
@jdbaldry - Can you confirm the changes to the `docs/sources/developers/http_api/correlations.md` file and the pinning versions strategy being used in this PR? I'm also not sure what the best option is for backporting here.

The other changes look fine to me.

File: docs/sources/developers/http_api/correlations.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 aliases:
   - ../../http_api/correlations/
-canonical: /docs/grafana/latest/developers/http_api/correlation/
+canonical: /docs/grafana/latest/developers/http_api/correlations/
```
Comment (jdbaldry) [inline docs/sources/developers/http_api/correlations.md:4]: Nice, thanks for fixing!

**Review by jdbaldry — COMMENTED** [review]
I haven't tested the links for resolution but I think the intent is fine. 

I wonder why we include the what's new for some many previous versions? I think maintaining those files over time is more work than it is probably worth.

**Review by brendamuir — APPROVED** [review]

**Comment by clementduveau** [comment]
> Do you know if the things you're fixing on these pages are a major pain point?

Unfortunately no. I am not sure we are tracking how many people clicks on internal links and end on a 404. Basically, even if we don't backport that, it's still nice to have to pin "What's new" links to the right version (Worst case: 404. Best case: you end on a page that is not related to your version, still not great)

I would say don't bother with previous versions of the "What's New" (docs versions I mean, not Grafana versions) by default, but again, no idea if people goes to previous versions of the doc to see "What's New" pages

File: docs/sources/alerting/difference-old-new.md [hunk 2]
```diff
@@ -13,13 +14,13 @@ labels:
     - cloud
     - enterprise
     - oss
-title: What's new in Grafana Alerting
+title: Grafana Alerting vs Legacy dashboard alerting
 weight: 108
 ---
 
 # What's new in Grafana Alerting
```
Comment (imatwawana) [inline docs/sources/alerting/difference-old-new.md:21]:
```suggestion
# Grafana Alerting vs Legacy dashboard alerting
```

**Comment by imatwawana** [comment]
@clementduveau - Thanks for the clarification. I've set it to backport a couple times and make sure it's in the current version of the docs as well and I'll go ahead and merge this.

**Pull request merged**